core/net/ip/tcpip.c

```diff
@@ -282,8 +294,8 @@ tcp_attach(struct uip_conn *conn,
   uip_tcp_appstate_t *s;
 
   s = &conn->appstate;
-  s->p = PROCESS_CURRENT();
-  s->state = appstate;
+
+  setup_appstate(s, appstate);
```

Comment:
In this function and the ones below it, the local variable `s` no longer is needed.
```c
  setup_appstate(&conn->appstate, appstate);
```